src/oic/oic/__init__.py

```diff
@@ -938,6 +939,12 @@ def do_user_info_request(
                 )
         elif resp.status_code == 500:
             raise PyoidcError("ERROR: Something went wrong: %s" % resp.text)
+        elif resp.status_code == 405:
```

Comment:
Have you thought about updating the general error handling (L948) to better handle this?

It looks to me that adding the status code to the thrown exception on L953, might be enough to put developers on the right track.

Reply:
Sorry for late reply, had to fix stuff for my study.
The problem with the general exception is that the ALLOW header exists only if a 405 is thrown.
Right now the exception a lib user gets is thrown at L953 (since the request is no valid json). To reach the
general error handling we would filter 405 out of the 400-500 range part.
This seems more complex to me than my solution.

Reply:
I was thinking about simply adding the `status_code` to the error message.

But this works as well.